// ngx-tui-image-editor/esm5/lib/toast-ui-image-editor.models.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
/**
 * @record
 */
export function IPosition() { }
/** @type {?} */
IPosition.prototype.x;
/** @type {?} */
IPosition.prototype.y;
/**
 * @record
 */
export function IAddTextEvent() { }
/** @type {?} */
IAddTextEvent.prototype.originPosition;
/** @type {?} */
IAddTextEvent.prototype.clientPosition;
/**
 * @record
 */
export function IMousedownEvent() { }
/** @type {?} */
IMousedownEvent.prototype.event;
/** @type {?} */
IMousedownEvent.prototype.originPointer;
/**
 * @record
 */
export function IGraphicObjectProps() { }
/** @type {?|undefined} */
IGraphicObjectProps.prototype.id;
/** @type {?|undefined} */
IGraphicObjectProps.prototype.type;
/** @type {?|undefined} */
IGraphicObjectProps.prototype.text;
/** @type {?|undefined} */
IGraphicObjectProps.prototype.left;
/** @type {?|undefined} */
IGraphicObjectProps.prototype.top;
/** @type {?|undefined} */
IGraphicObjectProps.prototype.width;
/** @type {?|undefined} */
IGraphicObjectProps.prototype.height;
/** @type {?|undefined} */
IGraphicObjectProps.prototype.fill;
/** @type {?|undefined} */
IGraphicObjectProps.prototype.stroke;
/** @type {?|undefined} */
IGraphicObjectProps.prototype.strokeWidth;
/** @type {?|undefined} */
IGraphicObjectProps.prototype.fontFamily;
/** @type {?|undefined} */
IGraphicObjectProps.prototype.fontSize;
/** @type {?|undefined} */
IGraphicObjectProps.prototype.fontStyle;
/** @type {?|undefined} */
IGraphicObjectProps.prototype.fontWeight;
/** @type {?|undefined} */
IGraphicObjectProps.prototype.textAlign;
/** @type {?|undefined} */
IGraphicObjectProps.prototype.textDecoration;
/** @type {?|undefined} */
IGraphicObjectProps.prototype.opacity;
/**
 * @record
 */
export function IOptions() { }
/** @type {?|undefined} */
IOptions.prototype.includeUI;
/** @type {?|undefined} */
IOptions.prototype.cssMaxWidth;
/** @type {?|undefined} */
IOptions.prototype.cssMaxHeight;
/** @type {?|undefined} */
IOptions.prototype.usageStatistics;
/** @type {?|undefined} */
IOptions.prototype.selectionStyle;
/**
 * @record
 */
function ISelectionStyleConfig() { }
/** @type {?|undefined} */
ISelectionStyleConfig.prototype.cornerStyle;
/** @type {?|undefined} */
ISelectionStyleConfig.prototype.cornerSize;
/** @type {?|undefined} */
ISelectionStyleConfig.prototype.cornerColor;
/** @type {?|undefined} */
ISelectionStyleConfig.prototype.cornerStrokeColor;
/** @type {?|undefined} */
ISelectionStyleConfig.prototype.transparentCorners;
/** @type {?|undefined} */
ISelectionStyleConfig.prototype.lineWidth;
/** @type {?|undefined} */
ISelectionStyleConfig.prototype.borderColor;
/** @type {?|undefined} */
ISelectionStyleConfig.prototype.rotatingPointOffset;
/**
 * @record
 */
function IIncludeUIOptions() { }
/** @type {?|undefined} */
IIncludeUIOptions.prototype.loadImage;
/** @type {?|undefined} */
IIncludeUIOptions.prototype.theme;
/** @type {?|undefined} */
IIncludeUIOptions.prototype.menu;
/** @type {?|undefined} */
IIncludeUIOptions.prototype.initMenu;
/** @type {?|undefined} */
IIncludeUIOptions.prototype.uiSize;
/** @type {?|undefined} */
IIncludeUIOptions.prototype.menuBarPosition;
/**
 * @record
 */
function IThemeConfig() { }
